router/core/header_rule_engine_test.go

```diff
@@ -241,3 +279,74 @@ func TestSubgraphNamedHeaderRule(t *testing.T) {
 	assert.Empty(t, updatedClientReq2.Header.Get("X-Test-Subgraph-1"))
 	assert.Equal(t, "Test-Value-2", updatedClientReq2.Header.Get("X-Test-Subgraph-2"))
 }
+
+func TestSubgraphReNamedHeaderRule(t *testing.T) {
+	ht, err := NewHeaderTransformer(config.HeaderRules{
+		Subgraphs: map[string]config.GlobalHeaderRule{
+			"subgraph-1": {
```

Comment:
Please separate the tests in smaller case like

- Rename named header
- Rename matched header
- Rename with default
- ...